[PRD-4896] SQL Query Designer Preview button not working

ricardosilva88 opened the pull request:
@tmorgner 

**Note for future reference (SP's process): This commit uses code made in context of https://jira.pentaho.com/browse/BACKLOG-23781 (already merged), please respect this order**

wingman-pentaho commented:
### Build Completed

:fire: This pull request has some issues. It would be preferable to fix them in order for it to be just perfect. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'designer/datasource-editor-jdbc,engine/core' -P '!assemblies' -Dsurefire.runOrder=alphabetical -Daudit -amd clean install
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'designer/datasource-editor-jdbc,engine/core' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  designer/datasource-editor-jdbc/src/main/java/org/pentaho/reporting/ui/datasources/jdbc/ui/JdbcQueryDesignerDialog.java
  engine/core/pom.xml
  engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/misc/datafactory/sql/SimpleSQLReportDataFactory.java
  engine/core/src/test/java/org/pentaho/reporting/engine/classic/core/modules/misc/datafactory/sql/SimpleSQLReportDataFactoryTest.java
```

#### Unit Test Coverage

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

<details>
<summary>:warning: <b>Coverage Changes</b> (click to expand)</summary>

###### org.pentaho.reporting.engine.classic.core.modules.misc.datafactory.sql.SimpleSQLReportDataFactory
 * Branch Change: **-1.2%** :small_red_triangle_down:
 * Complexity Change: **-2.1%** :small_red_triangle_down:

###### org.pentaho.reporting.designer.core.inspections.impl.InvalidQueryNameReferenceInspection
 * Branch Change: **-20.9%** :small_red_triangle_down:
 * Complexity Change: **-23.5%** :small_red_triangle_down:
 * Instruction Change: **-17.8%** :small_red_triangle_down:
 * Line Change: **-17.2%** :small_red_triangle_down:

</details>

tmorgner commented on engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/misc/datafactory/sql/SimpleSQLReportDataFactory.java:
```diff
@@ -130,9 +136,10 @@ protected synchronized Connection getConnection( final DataRow dataRow ) throws
-  private int getBestResultSetType( final DataRow dataRow ) throws SQLException {
-    if ( "simple".equalsIgnoreCase( getConfiguration().getConfigProperty( //$NON-NLS-1$
-        ResultSetTableModelFactory.RESULTSET_FACTORY_MODE ) ) ) { //$NON-NLS-1$
+  @VisibleForTesting
```
Is this Guava thing really necessary? This is a protected method that is only referenced from the same package (as the tests are local), so this should not be needed. Putting a new dependency in just for the sake of test annotations seems so wrong on so many levels. 

To Be honest, that method could even be public static, as the "globalConfig" variable you need from the local instance is actually fetched from a static context anyway. If you inline this, that becomes clear. And for static helper methods public does not break encapsulation so that is fine by all standards I'm aware of.

ricardosilva88 commented:
@tmorgner , just updated the PR. Removed the "Visible for testing" annotation, changed the getBestResultSetType method to public and globalConfig var to public static. Didn't put static on getBestResultSetType method because there are other methods that would be need a cascade update of static modifier, like getConnection.

ricardosilva88 commented:
@tmorgner  this PR depends on code made in context of https://jira.pentaho.com/browse/BACKLOG-23781, that wasn't fully merged, it is missing one PR - https://github.com/pentaho/pentaho-kettle/pull/5487

tmorgner commented:
Looks sane!
